sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala

```diff
@@ -135,6 +136,24 @@ object ShimLoader extends Logging {
       mutable.addURL(shimURL)
       mutable.addURL(shimCommonURL)
     }
+
+    if (pluginClassLoader == null) {
+      // Next we should check for the scala built in REPL interpreter class loader.
+      // This happens on the driver side as a part of the REPL. It does not use a
+      // MutableURLClassLoader, but has essentially done the same thing.
+      Option(contextClassLoader).foreach { loader =>
```

Comment:
LGTM, but I would generalize the pattern matching above and  make it another case to avoid have another place to add the same url and pluginClassLoader assingnment.

Reply:
I tried it, but it felt less clear and didn't save much. Did you mean you wanted to call addURL by reflection all of the time?

Reply:
along these lines,  unconditional by reflection was not my initial idea but I think it's reasonable here and will make the code even simpler.

Reply:
I tried and I ran into a few issues with using reflection for both paths. I don't really feel like spending more time to debug something that works just to save a few lines of code. So if you are okay with it I think I will just keep it as is.